### PR TITLE
kata_agent: Constraints the OCI spec

### LINKED ISSUE #559: kata_agent: Constraint the OCI spec
Passing the whole OCI spec provided through the config.json will not work because not everything will apply properly. Indeed, we are in a virtualized context, meaning that we have to limit what we provide to the agent.
Here is a list of things to do:
- [ ] `NEWNET` has to be removed from libcontainer `config.Namespaces`
- [ ] As a consequence of the previous point, we should not provide anything through `config.Networks`
- [ ] `config.Hooks` entry has to be `nil`
- [ ] `config.Seccomp` entry has to be `nil`
- [ ] `config.NoNewKeyring` has to be `true` since we don't want to force the kernel to support it
- [ ] `config.Mounts` has to be analyzed to remove unwanted mounts (TBD)

### PR DESCRIPTION
**Pull request by sboeuf**
This commit constraints the OCI spec provided to the agent on
several point such as seccomp, hooks, network namespace and
Intel RDT.

Fixes #559

Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>

### CODE REVIEW

File: kata_agent.go [hunk 1]
```diff
@@ -445,6 +445,22 @@ func (k *kataAgent) createContainer(pod *Pod, c *Container) error {
 	// We need to give the OCI spec our absolute rootfs path in the guest.
 	grpcSpec.Root.Path = rootPath
 
+	// Remove Hooks
+	grpcSpec.Hooks = nil
+
+	// Remove Seccomp
+	grpcSpec.Linux.Seccomp = nil
```
Comment (sameo) [inline kata_agent.go]: Any reason why we're removing SecComp? It should not be an issue to apply those inside the VM.

**Comment by sameo** [comment]
Besides the fact that I think we could keep seccomp in:
LGTM

File: kata_agent.go (hunk 1 above)
Comment (sboeuf) [reply kata_agent.go]: Unfortunately, we need several things to make the seccomp usable (as explained here https://github.com/kata-containers/agent/issues/104).
This is temporary until we can support seccomp with Kata Containers agent.

File: kata_agent.go [hunk 2]
```diff
@@ -445,6 +445,22 @@ func (k *kataAgent) createContainer(pod *Pod, c *Container) error {
...
+
+	// Remove Intel RDT
```
[the remark this answers is not in the document]
Comment (sboeuf) [reply kata_agent.go]: All done !

**Comment by sameo** [comment]
Now for real:
LGTM

**Pull request merged**